scrapy_onedrive_exporter/onedrive_exporter.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+import requests
+from scrapy.exceptions import NotConfigured
+from scrapy.extensions.feedexport import BlockingFeedStorage, build_storage
+
+logger = logging.getLogger(__name__)
+
+GRAPH_API_ENDPOINT = "https://graph.microsoft.com/v1.0"
+
+
+class OneDriveFeedStorage(BlockingFeedStorage):
+    def __init__(self, uri, access_token=None, *, feed_options=None):
+        if not access_token:
+            raise NotConfigured("Missing ONEDRIVE_ACCESS_TOKEN")
+
+        if feed_options and feed_options.get("overwrite", True) is False:
+            raise NotConfigured(
+                "OneDriveFeedStorage does not support appending to files. Please "
+                "remove the overwrite option from your FEEDS setting or set it to True."
+            )
+
+        self.session_headers = {
+            'Authorization': f'Bearer {access_token}',
+            'Content-Type': 'application/json'
+        }
+        self.file_path = uri.replace('onedrive://', '')
+        self.session_url = self._create_session(self.file_path)
+
+    @classmethod
+    def from_crawler(cls, crawler, uri, *, feed_options=None):
+        return build_storage(
+            cls,
+            uri,
+            access_token=crawler.settings.get("ONEDRIVE_ACCESS_TOKEN"),
+            feed_options=feed_options,
+        )
+
+    def _create_session(self, file_path):
+        session_endpoint = f"/me/drive/root:/{file_path}:/createUploadSession"
+        response = requests.post(GRAPH_API_ENDPOINT + session_endpoint, headers=self.session_headers)
+        try:
+            return response.json()['uploadUrl']
+        except Exception as e:
+            raise NotConfigured(
+                f"Failed to create the upload Session: {e}. Response: {response.content}"
+            )
+
+    def _store_in_thread(self, file):
+        file.seek(0)
+        content = file.read()
+        content_range = f"bytes 0-{len(content) - 1}/{len(content)}"
+        headers = {'Content-Length': str(len(content)), 'Content-Range': content_range}
```

Comment:
Do we need to specify `Content-Length`? I would expect `requests` to set that one automatically, although maybe it does not if you set `Content-Range` as well.

Reply:
@Gallaecio I tried removing it and it worked, please check again. thanks!